### PR TITLE
Let the caller to set the build api credentials.

### PR DESCRIPTION
**Pull request by ser-io**

### CODE REVIEW

File: frontend/src/libhoclient/host_orchestrator_client.go [hunk 1]
```diff
@@ -48,11 +48,26 @@ type ConnectWebRTCOpts struct {
...
 	AccessToken   string
 	UserProjectID string
 }
 
+func (c *AccessTokenBuildAPICreds) ApplyToHTTPRequest(rb *HTTPRequestBuilder) {
+	const HTTPHeaderBuildAPICreds = "X-Cutf-Host-Orchestrator-BuildAPI-Creds"
+	const HTTPHeaderBuildAPICredsUserProjectID = "X-Cutf-Host-Orchestrator-BuildAPI-Creds-User-Project-ID"
+	if c.AccessToken != "" {
+		rb.AddHeader(HTTPHeaderBuildAPICreds, c.AccessToken)
+		if c.UserProjectID != "" {
```
Comment (jemoreira) [inline frontend/src/libhoclient/host_orchestrator_client.go:65]: what happens if `AccessToken` is empty and `UserProjectID` is not?

File: frontend/src/libhoclient/host_orchestrator_client.go [hunk 2]
```diff
@@ -48,11 +48,26 @@ type ConnectWebRTCOpts struct {
 	LocalICEConfig *wclient.ICEConfig
 }
 
-type BuildAPICredential struct {
+type BuildAPICreds interface {
+	ApplyToHTTPRequest(rb *HTTPRequestBuilder)
+}
+
+type AccessTokenBuildAPICreds struct {
```
Comment (jemoreira) [inline frontend/src/libhoclient/host_orchestrator_client.go:55]: What about the one that asks the Cloud Orchestrator to inject? Are you going to implement it in the cloud orchestrator project?

**Review by jemoreira — APPROVED** [review]

File: frontend/src/libhoclient/host_orchestrator_client.go (hunk 1 above)
Comment (ser-io) [reply frontend/src/libhoclient/host_orchestrator_client.go:65]: the `UserProjectID` value is only relevant if there's an access token.

File: frontend/src/libhoclient/host_orchestrator_client.go (hunk 2 above)
Comment (ser-io) [reply frontend/src/libhoclient/host_orchestrator_client.go:55]: yes.

**Pull request merged**